tests/gcm/test_unit_change.py

```diff
@@ -0,0 +1,94 @@
+import numpy as np
+import pandas as pd
+import pytest
+from flaky import flaky
+from sklearn.exceptions import NotFittedError
+from sklearn.linear_model import LinearRegression
+from sklearn.svm import SVR
+
+from dowhy.gcm.unit_change import SklearnLinearMechanism, SklearnMechanism, unit_change_linear, unit_change_nonlinear
+
+
+@flaky(max_runs=5)
+def test_given_fitted_linear_mechanisms_with_output_change_when_evaluate_unit_change_linear_then_returns_expected_result():
+    num_rows = 100
+    A1 = np.random.normal(size=num_rows)
+    B1 = np.random.normal(size=num_rows)
+    C1 = 2 * A1 + 3 * B1
+
+    A2 = np.random.normal(size=num_rows)
+    B2 = np.random.normal(size=num_rows)
+    C2 = 3 * A2 + 2 * B2
+
+    background_df = pd.DataFrame(data=dict(A=A1, B=B1))
+    foreground_df = pd.DataFrame(data=dict(A=A2, B=B2))
+
+    background_mechanism = SklearnLinearMechanism(
+        LinearRegression(fit_intercept=False).fit(np.column_stack((A1, B1)), C1)
+    )
+    foreground_mechanism = SklearnLinearMechanism(
+        LinearRegression(fit_intercept=False).fit(np.column_stack((A2, B2)), C2)
+    )
+
+    actual_contributions = unit_change_linear(
+        background_mechanism, background_df, foreground_mechanism, foreground_df, ["A", "B"]
+    )
+    expected_contributions = pd.DataFrame(
+        data=dict(
+            A=(3 + 2) * (A2 - A1) / 2, B=(2 + 3) * (B2 - B1) / 2, f=(A1 + A2) * (3 - 2) / 2 + (B1 + B2) * (2 - 3) / 2
+        )
+    )
```

Comment:
Actually, here it would help to put the dict pairs on separate lines. Feel free to omit `data=` and put `dict` on the same line as `DataFrame` to reduce vertical space.

Reply:
Black does not like that suggestion.

Reply:
Well then...